actor/monk_ki.js

```diff
@@ -0,0 +1,231 @@
+        return
+    }
+
+    let monk = actor.items.find(i => i.name === 'Monk' && i.type === 'class');
+    if (!monk) {
+        ui.notifications.warn(errNoMonkToken);
+        return
+    }
+
+    let monkLevels = monk.data.data.levels || 20;
+    //let subClass = monk.data.data.subclass;
+
+    if (monkLevels < 2) {
+        ui.notifications.warn('You must have a least 2 Monk levels to use ki points.');
+        return
+    }
+
+    class KiFeature {
+        /**
+         * @param {string} name
+         * @param {string} fallbackText
+         * @param {number} requireLevel
+         * @param {function} action
+         * @param {function} appendTemplate
+         * @param {number} kiCost
+         */
+        constructor(name, fallbackText, requireLevel, action, appendTemplate, kiCost) {
+            this.name = name;
+            this.fallbackText = fallbackText;
+            this.requireLevel = requireLevel;
+            this.kiCost = kiCost || 1;
+            if (action) {
+                this.action = action;
+            }
+            if (appendTemplate) {
+                this.appendTemplate = appendTemplate;
+            }
+        }
+
+        render(allowHigher) {
+            let entry = null;
+
+            const pack = game.packs.get("dnd5e.classfeatures");
+            if (!pack) {
+                console.warn('Could not find "dnd5e.classfeatures" compendium.');
+            } else {
+                entry = pack.index.find(e => e.name === this.name);
+            }
+
+            if (!allowHigher && this.requireLevel && monkLevels && this.requireLevel > monkLevels) {
+                ui.notifications.warn(`You need to have ${this.requireLevel} monk levels, you only have ${monkLevels}.`)
+                return
+            }
+
+            if (entry) {
+                pack.getEntity(entry._id).then(o => {
+                    let template = `@Compendium[dnd5e.classfeatures.${entry._id}]{${this.name}}
+                    ${o.data.data.description.value}`;
+                    if (this.appendTemplate) {
+                        template += '\n\n' + this.appendTemplate();
+                    }
+                    sendChat(template);
+                    if (this.action) {
+                        this.action();
+                    }
+                });
+            } else {
+                console.warn(`Could not find "${this.name}" entry in compendium.`);
+                let template = this.fallbackText;
+                if (this.appendTemplate) {
+                    template += '\n\n' + this.appendTemplate();
+                }
+                sendChat(template);
+                if (this.action) {
+                    this.action();
+                }
+            }
+        }
+    }
+
+    const features = [
+        new KiFeature("Ki: Flurry of Blows",
+            "Immediately after you take the <b>Attack</b> action on your turn, you can spend 1 ki point to make two unarmed strikes as a bonus action.",
```

Comment:
There is probably a better way to also do this, but would be nice to include open hand technique if the user has it.

```suggestion

   const openHand = !!actor.items.find(o => o.data.name === 'Open Hand Technique') ? `<br />In addition, you can impose one of the following: <ul><li>It must succeed on a <b>Dexterity</b> saving throw or be knocked prone.</li><li>It must make a <b>Strength</b> saving throw. If it fails, you can push it up to 15 feet away from you.</li><li>It can’t take reactions until the end of your next turn.</li></ul> Saving throw <b>DC ${10 + actor.data.data.abilities.wis.mod}</b>` : "";
console.log(!!actor.items.find(o => o.data.name === 'Open Hand Technique'));

    const features = [
        new KiFeature("Ki: Flurry of Blows",
            `Immediately after you take the <b>Attack</b> action on your turn, you can spend 1 ki point to make two unarmed strikes as a bonus action. ${openHand}`,
```